### PR TITLE
ref: Replace failure with anyhow in symsorter

### PR DESCRIPTION
**Pull request by Swatinem**
Also updated symbolic dependency version because of the lifetime inference problem.

#skip-changelog

### CODE REVIEW

**Comment by jan-auer** [comment]
You can shim `symbolic`'s errors similar to:

https://github.com/getsentry/relay/blob/bbb063c512ee113986d12117c49577e4635c75bc/tools/scrub-minidump/src/main.rs#L80-L83

File: symsorter/src/app.rs [hunk 1]
```diff
@@ -124,13 +121,15 @@ fn process_file(
...
     };
-    let archive = maybe_ignore_error!(Archive::parse(&bv));
+    let archive = maybe_ignore_error!(
+        Archive::parse(&bv).map_err(|e| anyhow!("failed to parse archive {}", e))
+    );
     let root = &RunConfig::get().output;
 
     for obj in archive.objects() {
-        let obj = maybe_ignore_error!(obj);
+        let obj = maybe_ignore_error!(obj.map_err(|e| anyhow!(e)));
```
Comment (jan-auer) [inline symsorter/src/app.rs:130]: NB: We need to remember to come back to these and replace them with `.context("xyz")` once we get `symbolic` fixed.

File: symsorter/src/app.rs [hunk 2]
```diff
@@ -335,9 +332,9 @@ pub fn main() -> ! {
         Ok(()) => std::process::exit(0),
         Err(error) => {
             eprintln!("{}: {}", style("error").red().bold(), error);
-            for cause in error.iter_causes() {
+            error.chain().skip(1).for_each(|cause| {
```
Comment (jan-auer) [inline symsorter/src/app.rs:335]: Let's stick to the regular `for` syntax:

```suggestion
            for cause in error.chain().skip(1) {
```

File: symsorter/src/utils.rs [hunk 3]
```diff
@@ -54,21 +54,22 @@ pub fn get_target_filename(obj: &Object) -> Option<PathBuf> {
...
-            if writer.write_object(
-                &obj,
-                &path.file_name().unwrap().to_string_lossy().to_string(),
-            )? {
+            let writer =
+                SourceBundleWriter::start(Cursor::new(&mut out)).map_err(|e| anyhow!(e))?;
+            if writer
+                .write_object(
+                    &obj,
+                    &path.file_name().unwrap().to_string_lossy().to_string(),
```
Comment (jan-auer) [inline symsorter/src/utils.rs:69]: _nit_: While we're at it, this could be moved into a temporary variable for readability.

**Review by jan-auer — APPROVED** [review]

File: symsorter/src/utils.rs (hunk 3 above)
Comment (Swatinem) [reply symsorter/src/utils.rs]: also the `to_string` is useless there ;-)

**Pull request merged**